airbyte-integrations/connectors/source-pipedrive/source_pipedrive/streams.py

```diff
@@ -77,9 +77,14 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
         """
         :return an iterable containing each record in the response
         """
+        documented_properties = self.get_json_schema()["properties"]
         records = response.json().get(self.data_field) or []
         for record in records:
             record = record.get(self.data_field) or record
+            # do not emit non-documented top level fields as part of a record
+            if isinstance(record, dict):
+                # it can also be a list, so need to handle it
+                record = {key: value for (key, value) in record.items() if key in documented_properties}
```

Comment:
Why do we need to restrict this in the source? This is something the platform will be taking care of. If the fields are not documented, we should document them instead of not emitting them. See [this issue](https://github.com/airbytehq/airbyte/issues/24493)

Reply:
That would be too easy :smile: 
This connector's build failed due to the fields that are not even documented by the source API (Pipedrive)

Reply:
I don't think we should document and expose fields that are not documented by Pipedrive

Reply:
Oh... that is indeed sort of yikes 🤔 What types of info are they? Is there anything in the github issue that I made that _is_ documented by pipedrive? At this point I would rather write a distinct list of columns to ignore (that we know aren't documented by pipedrive) instead of always non-emitting ones that we haven't declared. That makes us unresilient to new (documented) things in the pipedrive API - we won't notice them nor emit them.

Reply:
and the data type of those fields varies a lot - strings, integers, objects etc

Reply:
> new undocumented fields - we won't notice them in the output as well

we will, actually! They will appear on the connector problems dashboard under [Record Unexpected Fields](https://app.datadoghq.com/dashboard/hq4-ush-fht/connector-problems?from_ts=1683044883832&to_ts=1683649683832&live=true)

> And hardcoding things like 3ce5b1409718d65a8adc965be1f0da8821d8b9ab (from your issue) makes me a bit nervous 😸

Yeah I totally get this, I wonder if we can do something specific for that stream. 

> and the data type of those fields varies a lot - strings, integers, objects etc

Not sure exactly what you're getting at here - for a given field does it differ? Or all the fields have different types? 


Let me see if other people have opinions on how we want to handle fields undocumented by the API in general

Reply:
> Not sure exactly what you're getting at here - for a given field does it differ? Or all the fields have different types?

that was just an answer to your question - "What types of info are they?"

Reply:
Ah, I didn't mean `type`, just more meant "what information are they sending through that they're not documenting"

Reply:
@erohmensing did you have a chance to know what people think of this?